### ISSUE TITLE
Error handling and robustness

### ISSUE THREAD

**Issue by mg0536** [rudders/homebridge-platform-wemo#51]
First off - thanks for the really great work here. I'm so happy to have found your code and use it in my home.

I'm seeing some weirdness when trying to setup up homekit triggers on iOS, using the selfcoded.com (Matthias Hochgatterer) "Home" app.  

When operating the wemo outlet switch services manually, again using Home, I sometimes see "error" instead of "on" or '"off" for switch status. If I soldier on, ignoring the error and operate the toggle in Home, the switch usually operates, and resets to a known state; I'm almost always successful after a second try. It seems like single polls can fail, perhaps due to wemo protocol or connectivity issues, but overall, after a small number of retries, they succeed.

The issue elevates to a significant problem when I try to add triggers, again using Home on iOS. It seems like Home will not poll again if there is an error with a service, and further, will not run a trigger associated with a scene which includes the service reporting an error. Error status persists, even if the failure is intermittent. The result is that triggers almost always fail to run (2/28 triggers ran in the last 2 weeks). 

I'm not sure where the underlying bug is. I'm guessing it's in the wemo protocol itself, but perhaps this can be addressed within homebridge-platform-wemo? I'm imagining a certain number of retries within the app, before reporting an error, when polled.

My simple setup: 3 Wemo outlet switches, homebridge and homebridge-platform-wemo at the latest versions, running via launchd on my Mac. "Home" app running on iPhone and iPad.  Mac OS and iOS at the latest versions. Home at the latest version, as of August 17, 2016.

Thanks again for your great work getting Wemo to play with HomeKit,

-Matt

**Comment by driskell** [comment]
If it helps, I encountered this with Home before with a `set` callback that returned too early. What sometimes happens is you hit the button in Home and it sends the request, the `set` code then runs, but if you return before the process completes I *THINK* that Home sends a `get` and because it's not yet actually finished setting it sees the state is incorrect and throws an error bubble.

I can see in the code that the `set` doesn't wait for the process to complete before returning so it could be that which causes it. Calling the callback only when the set completes might resolve this.

**Comment by mg0536** [comment]
Here's an update = I've noticed drastically improved performance in regard to error handling and robustness:

1. I'm running homebridge on a wifi connected Mac Mini. It was dropping the wifi link VERY frequently, when I wasn't sitting in front of it. I added a wifi persist script to bring it up, if it's down. Works much better now.

2. Upgraded to iOS 10 - using Apple's "Home" app seems to work better than the selfcoded.com (Matthias Hochgatterer) "Home" app.  

Not sure which one of these changes matters more, but my Wemo outlets seem to be much more responsive, now. I've only had a few "didn't hear back" errors from Siri in the past week.

Thanks again for homebridge - really enjoy having it in my home.

-Matt
